### PR TITLE
React

### PR DESCRIPTION
**Pull request by michellenicolephillips**

### CODE REVIEW

File: website/src/components/p5js-components/magic8.js [hunk 1]
```diff
@@ -0,0 +1,88 @@
+import React from 'react';
+import p5 from 'p5';
+import '../projects.css';
+
+
+class Magic8 extends React.Component {
+    constructor(props) {
+        super(props)
+        this.myRef = React.createRef()
```
Comment (michellenicole23) [inline website/src/components/p5js-components/magic8.js:9]: helo!

File: .eslintcache [hunk 2]
```diff
@@ -0,0 +1 @@
...c/reportWebVitals.js",[],"/Users/Michelle/Documents/GitHub/portfolio/src/components/nav_bar_component.js",[],"/Users/Michelle/Documents/GitHub/portfolio/src/components/footer.js",[],"/Users/Michelle/Documents/GitHub/portfolio/src/components/home.js",[],"/Users/Michelle/Documents/GitHub/portfolio/src/components/resume.js",[],"/Users/Michelle/Documents/GitHub/portfolio/src/components/projects.js",[],"/Users/Michelle/Documents/GitHub/portfolio/src/components/p5js-components/rain.js",[],"/Users/Michelle/Documents/GitHub/portfolio/src/components/p5js-components/bookshelf.js",[],"/Users/Michelle/Documents/GitHub/portfolio/src/components/p5js-components/house.js",[],"/Users/Michelle/Documents/GitHub/portfolio/src/components/p5js-components/fishtank.js",[],"/Users/Michelle/Documents/GitHub/portfolio/src/components/p5js-components/magic8.js",[],"/Users/Michelle/Documents/GitHub/portfolio/src/components/p5js-components/animal.js",[],"/Users/Michelle/Documents/GitHub/portfolio/src/components/p5js-components/city.js",[],"/Users/Michelle/Documents/GitHub/portfolio/src/components/p5js-components/spork.js",[],"/Users/Michelle/Documents/GitHub/portfolio/src/components/p5js-components/plate.js",[]]
```
Comment (kushagharahi) [inline .eslintcache]: add this file to your gitignore (but research to make sure that's the right move first)

File: .vscode/settings.json [hunk 3]
```diff
@@ -0,0 +1,3 @@
+{
+    "git.ignoreLimitWarning": true
+}
```
Comment (kushagharahi) [inline .vscode/settings.json]: gitignore your .vscode folder as it relates to your vscode not your project

Comment (kushagharahi) [reply .vscode/settings.json]: also delete it from your git cache to git rid of it here (google how to do this)

File: package.json [hunk 4]
```diff
@@ -0,0 +1,41 @@
...
+    "@testing-library/jest-dom": "^5.11.9",
+    "@testing-library/react": "^11.2.3",
+    "@testing-library/user-event": "^12.6.2",
+    "p5": "^1.2.0",
+    "react": "^17.0.1",
+    "react-dom": "^17.0.1",
+    "react-p5-wrapper": "^2.3.0",
+    "react-router-dom": "^5.2.0",
+    "react-scripts": "4.0.1",
+    "web-vitals": "^0.2.4"
```
Comment (kushagharahi) [inline package.json:15]: delete your project and redownload it to make sure all your dependencies are captured correctly

File: public/index.html [hunk 5]
```diff
@@ -0,0 +1,42 @@
+<!DOCTYPE html>
+<html lang="en">
+  <head>
+    <meta charset="utf-8" />
+    <link rel="icon" href="%PUBLIC_URL%/logo.ico" />
+    <meta name="viewport" content="width=device-width, initial-scale=1" />
+    <meta name="theme-color" content="#000000" />
+    <meta
+      name="description"
+      content="Web site created using create-react-app"
```
Comment (kushagharahi) [inline public/index.html]: you'll want to edit this to fit your website

File: public/index.html [hunk 6]
```diff
@@ -0,0 +1,42 @@
+<!DOCTYPE html>
+<html lang="en">
+  <head>
+    <meta charset="utf-8" />
+    <link rel="icon" href="%PUBLIC_URL%/logo.ico" />
```
Comment (kushagharahi) [inline public/index.html:5]: this is how you set your favicon

File: public/index.html [hunk 7]
```diff
@@ -0,0 +1,42 @@
...
+    <meta name="viewport" content="width=device-width, initial-scale=1" />
+    <meta name="theme-color" content="#000000" />
+    <meta
+      name="description"
+      content="Web site created using create-react-app"
+    />
+    <link rel="apple-touch-icon" href="%PUBLIC_URL%/logo192.png" />
```
Comment (kushagharahi) [inline public/index.html:12]: is this needed?

Comment (kushagharahi) [reply public/index.html:12]: if not delete the file related to this too

File: public/manifest.json [hunk 8]
```diff
@@ -0,0 +1,25 @@
+{
+  "short_name": "React App",
+  "name": "Create React App Sample",
```
Comment (kushagharahi) [inline public/manifest.json]: you may want to edit this file to fit your website

File: src/components/footer.js [hunk 9]
```diff
@@ -0,0 +1,20 @@
...
+
+class Footer extends React.Component {
+  render() {
+    return ( 
+        <footer className="footer"> 
+            <p> Check me out on:<br/> 
+                <a className="social" href="https://github.com/michellenicole23" target="_blank" rel="noreferrer"><img src= {githublogo} alt="github"/></a>
+                <a className="social" href="https://www.linkedin.com/in/michelle-phillips-8a56a764/" target="_blank" rel="noreferrer" ><img src={linkedinlogo} alt="linkedin"/></a>
+            </p>
+    </footer>
```
Comment (kushagharahi) [inline src/components/footer.js]: formatting is off, tab this over

File: src/components/footer.js [hunk 10]
```diff
@@ -0,0 +1,20 @@
+import React from 'react';
+import './footer.css';
+import githublogo from '../res/images/github.png';
+import linkedinlogo from '../res/images/linkedin.png';
+
+
```
Comment (kushagharahi) [inline src/components/footer.js]: too many carriage returns here

File: src/components/home.js [hunk 11]
```diff
@@ -0,0 +1,35 @@
+import React from 'react';
+import './home.css';
+
+
+
+class Home extends React.Component {
```
Comment (kushagharahi) [inline src/components/home.js:4]: too many carriage returns here

File: src/components/nav_bar_component.js [hunk 12]
```diff
@@ -0,0 +1,41 @@
...
+import Projects from './projects';
+
+class NavBar extends React.Component {
+  render() {
+    return ( 
+    <Router>
+      <div>
+        <ul className="navbar">
+            <li className="navlink"><NavLink exact to="/" activeClassName="active">Home</NavLink></li>
+            <li className="navlink"><NavLink to="/projects" activeClassName="active">Projects</NavLink></li>
```
Comment (kushagharahi) [inline src/components/nav_bar_component.js]: make the rest of these `exact to` as well to be consistent

File: src/components/nav_bar_component.js [hunk 13]
```diff
@@ -0,0 +1,41 @@
...
+            <Home />
+          </Route>
+        </Switch>
+      </div>
+    </Router>
+    );
+  }
+}
+
+
```
Comment (kushagharahi) [inline src/components/nav_bar_component.js]: too many carriage returns here

File: src/components/p5js-components/animal.js [hunk 14]
```diff
@@ -0,0 +1,83 @@
+import React from 'react';
+import p5 from 'p5';
+import '../projects.css';
+
+class Animal extends React.Component {
+    constructor(props) {
+        super(props)
+        this.myRef = React.createRef()
+      }
```
Comment (kushagharahi) [inline src/components/p5js-components/animal.js]: too tabbed over

File: src/components/p5js-components/animal.js [hunk 15]
```diff
@@ -0,0 +1,83 @@
...
+            sketch.line(faceX*38/20, faceY*1.25, faceX*21/20, faceY);//bottomright
+            
+            sketch.fill(189, 144, 187);//nose color
+            sketch.ellipse(faceX, faceY, noseSize, noseSize);//nose
+    
+            sketch.fill(100, 148, 217);//gum color
+            sketch.ellipse(faceX, faceY+30, gumSize, gumSize);//gum
+            if (gumSize > faceSize){
+                gumSize -= faceSize
+            }else {
```
Comment (kushagharahi) [inline src/components/p5js-components/animal.js]:
```suggestion
            } else {
```

File: src/components/p5js-components/animal.js [hunk 16]
```diff
@@ -0,0 +1,83 @@
...
+            gumSize = gumSize+1;
```
Comment (kushagharahi) [inline src/components/p5js-components/animal.js]: tab  this over

File: src/components/p5js-components/animal.js [hunk 17]
```diff
@@ -0,0 +1,83 @@
...
+            sketch.line(faceX*38/20, faceY*0.75, faceX*21/20, faceY);//topright
+            sketch.line(faceX*38/20, faceY*1.25, faceX*21/20, faceY);//bottomright
+            
+            sketch.fill(189, 144, 187);//nose color
+            sketch.ellipse(faceX, faceY, noseSize, noseSize);//nose
+    
+            sketch.fill(100, 148, 217);//gum color
+            sketch.ellipse(faceX, faceY+30, gumSize, gumSize);//gum
+            if (gumSize > faceSize){
+                gumSize -= faceSize
```
Comment (kushagharahi) [inline src/components/p5js-components/animal.js]:
```suggestion
                gumSize -= faceSize;
```

File: src/components/p5js-components/animal.js [hunk 18]
```diff
@@ -0,0 +1,83 @@
+import React from 'react';
+import p5 from 'p5';
+import '../projects.css';
+
+class Animal extends React.Component {
+    constructor(props) {
+        super(props)
+        this.myRef = React.createRef()
```
Comment (kushagharahi) [inline src/components/p5js-components/animal.js]:
```suggestion
        this.myRef = React.createRef();
```

File: src/components/p5js-components/animal.js [hunk 19]
```diff
@@ -0,0 +1,83 @@
+import React from 'react';
+import p5 from 'p5';
+import '../projects.css';
+
+class Animal extends React.Component {
+    constructor(props) {
+        super(props)
```
Comment (kushagharahi) [inline src/components/p5js-components/animal.js]:
```suggestion
        super(props);
```

File: src/components/p5js-components/animal.js [hunk 20]
```diff
@@ -0,0 +1,83 @@
...
+    
+            sketch.fill(100, 148, 217);//gum color
+            sketch.ellipse(faceX, faceY+30, gumSize, gumSize);//gum
+            if (gumSize > faceSize){
+                gumSize -= faceSize
+            }else {
+            gumSize = gumSize+1;
+            }
+          };
+        });
```
Comment (kushagharahi) [inline src/components/p5js-components/animal.js]: are these ending curly brackets in line with the opening brackets? I can't tell in this PR

File: src/components/p5js-components/animal.js [hunk 21]
```diff
@@ -0,0 +1,83 @@
...
+
+
+  componentDidMount() {
+    this.myP5 = new p5(this.sketch, this.myRef.current)
```
Comment (kushagharahi) [inline src/components/p5js-components/animal.js]:
```suggestion
    this.myP5 = new p5(this.sketch, this.myRef.current);
```

File: src/components/p5js-components/animal.js [hunk 22]
```diff
@@ -0,0 +1,83 @@
...
+
+    render() {
+      return (
+          <div className="projects">
+            <div ref={this.myRef}>
+            </div>
+            <p className="project-info">    The third project was more practice using variables, especially for any repetitve information such as positions or size.
+                    The project was called "Animal Attack" and was originally the mouse's nose growing too big for the screen.
+                    For this site, I have added an "If/Else" statement to repeat the animation. It is now a mouse blowing bubble gum.</p>
+        </div>
```
Comment (kushagharahi) [inline src/components/p5js-components/animal.js]: tab this over

File: src/components/p5js-components/animal.js [hunk 23]
```diff
@@ -0,0 +1,83 @@
...
+  componentDidMount() {
+    this.myP5 = new p5(this.sketch, this.myRef.current)
+  }
+
+    render() {
+      return (
+          <div className="projects">
+            <div ref={this.myRef}>
+            </div>
+            <p className="project-info">    The third project was more practice using variables, especially for any repetitve information such as positions or size.
```
Comment (kushagharahi) [inline src/components/p5js-components/animal.js]:
```suggestion
            <p className="project-info">The third project was more practice using variables, especially for any repetitive information such as positions or size.
```

File: src/components/p5js-components/bookshelf.js [hunk 24]
```diff
@@ -1,4 +1,16 @@
-let bookshelf = new p5(( sketch ) => {
+import React from 'react';
+import p5 from 'p5';
+import '../projects.css';
+import starImage from '../../res/images/Star.png';
+
+class Bookshelf extends React.Component {
+    constructor(props) {
+    super(props)
```
Comment (kushagharahi) [inline src/components/p5js-components/bookshelf.js]:
```suggestion
    super(props);
```

File: src/components/p5js-components/bookshelf.js [hunk 25]
```diff
@@ -1,4 +1,16 @@
...
+    this.myRef = React.createRef()
```
Comment (kushagharahi) [inline src/components/p5js-components/bookshelf.js]:
```suggestion
    this.myRef = React.createRef();
```

File: src/components/p5js-components/bookshelf.js [hunk 26]
```diff
@@ -111,4 +141,22 @@ let bookshelf = new p5(( sketch ) => {
         }
         sketch.fill(108, 157, 240);            
         };
-    }, 'bookshelf');
\ No newline at end of file
+    };
+    componentDidMount() {
+        this.myP5 = new p5(this.Sketch, this.myRef.current)
+      }
```
Comment (kushagharahi) [inline src/components/p5js-components/bookshelf.js]: ending bracket formatting

File: src/components/p5js-components/bookshelf.js [hunk 27]
```diff
@@ -111,4 +141,22 @@ let bookshelf = new p5(( sketch ) => {
         }
         sketch.fill(108, 157, 240);            
         };
-    }, 'bookshelf');
\ No newline at end of file
+    };
+    componentDidMount() {
+        this.myP5 = new p5(this.Sketch, this.myRef.current)
```
Comment (kushagharahi) [inline src/components/p5js-components/bookshelf.js]:
```suggestion
        this.myP5 = new p5(this.Sketch, this.myRef.current);
```

File: src/components/p5js-components/city.js [hunk 28]
```diff
@@ -0,0 +1,76 @@
+import React from 'react';
+import p5 from 'p5';
+import '../projects.css';
+
+
+class City extends React.Component {
```
Comment (kushagharahi) [inline src/components/p5js-components/city.js:5]: too many carriage returns

File: src/components/p5js-components/city.js [hunk 29]
```diff
@@ -0,0 +1,76 @@
...
+    constructor(props) {
+        super(props)
```
Comment (kushagharahi) [inline src/components/p5js-components/city.js]:
```suggestion
        super(props);
```

File: src/components/p5js-components/city.js [hunk 30]
```diff
@@ -0,0 +1,76 @@
...
+        this.myRef = React.createRef()
```
Comment (kushagharahi) [inline src/components/p5js-components/city.js]:
```suggestion
        this.myRef = React.createRef();
```

File: src/components/p5js-components/city.js [hunk 31]
```diff
@@ -0,0 +1,76 @@
...
+          sketch.background(29, 40, 115);
+      
+          sketch.fill(255, 242, 0);
+          sketch.ellipse(xPos, yPos, 10, 10);//star 1
+          sketch.ellipse(xPos+50, yPos, 10, 10);//star 2
+          sketch.ellipse(xPos+100, yPos, 10, 10);//star 3
+          sketch.ellipse(xPos+150, yPos, 10, 10);//star 4
+          sketch.ellipse(xPos+200, yPos, 10, 10);//star 5
+          //reset star movement
+          if (yPos >= height){
```
Comment (kushagharahi) [inline src/components/p5js-components/city.js]:
```suggestion
          if (yPos >= height) {
```
consistency! if you're going to put a space before the bracket, do it everywhere. if you're not going to have a space, do it everywhere!

File: src/components/p5js-components/city.js [hunk 32]
```diff
@@ -0,0 +1,76 @@
...
+            xPos -= width;
+          } else {
+            xPos+=5;
+            yPos+=5;
+          }
+          sketch.fill(255, 255, 255);
+          sketch.ellipse(xPlane+10, yPlane,70, 25);//plane body
+          sketch.ellipse(xPlane+20, yPlane, 20, 60);//plane wings
+          //reset plane movement
+          if (xPlane >= width){
```
Comment (kushagharahi) [inline src/components/p5js-components/city.js]: needs to be consistent

File: src/components/p5js-components/city.js [hunk 33]
```diff
@@ -0,0 +1,76 @@
...
+            xPlane -= width
```
Comment (kushagharahi) [inline src/components/p5js-components/city.js]:
```suggestion
            xPlane -= width;
```

File: src/components/p5js-components/city.js [hunk 34]
```diff
@@ -0,0 +1,76 @@
...
+          } else {
+          xPlane++}
```
Comment (kushagharahi) [inline src/components/p5js-components/city.js]: tab, bracket on new line. formatting love needed

File: src/components/p5js-components/city.js [hunk 35]
```diff
@@ -0,0 +1,76 @@
...
+      
+          sketch.fill(0, 0, 0);//buildings
+          sketch.rect(0, 300, 50, 100);//building 1
+          sketch.rect(50, 250, 75, 150);//building 2
+          sketch.rect(130,300, 45, 100);//building 3
+          sketch.rect(175,200, 50, 200);// building 4
+          sketch.rect(225, 250, 50, 150);// building 5
+          sketch.rect(275, 325, 75, 75);//building 6
+        };
+        };
```
Comment (kushagharahi) [inline src/components/p5js-components/city.js]: lots of formatting love needed in this file. mistabbed, not neat.

File: src/components/p5js-components/city.js [hunk 36]
```diff
@@ -0,0 +1,76 @@
...
+        componentDidMount() {
```
Comment (kushagharahi) [inline src/components/p5js-components/city.js]: carriage return needed

File: src/components/p5js-components/city.js [hunk 37]
```diff
@@ -0,0 +1,76 @@
...
+            this.myP5 = new p5(this.Sketch, this.myRef.current)
+          }
```
Comment (kushagharahi) [inline src/components/p5js-components/city.js]: carriage return needed

File: src/components/p5js-components/city.js [hunk 38]
```diff
@@ -0,0 +1,76 @@
...
+          sketch.rect(0, 300, 50, 100);//building 1
+          sketch.rect(50, 250, 75, 150);//building 2
+          sketch.rect(130,300, 45, 100);//building 3
+          sketch.rect(175,200, 50, 200);// building 4
+          sketch.rect(225, 250, 50, 150);// building 5
+          sketch.rect(275, 325, 75, 75);//building 6
+        };
+        };
+        componentDidMount() {
+            this.myP5 = new p5(this.Sketch, this.myRef.current)
```
Comment (kushagharahi) [inline src/components/p5js-components/city.js]: either use semicolons in your js or don't. don't use them and not use them. consistency!

File: src/components/p5js-components/city.js [hunk 39]
```diff
@@ -0,0 +1,76 @@
...
+                <p class="project-info">
+                    The second project was to show my new knowledge of variables and basic animation skills.
+                    When transferring it to my portfolio, I added in "If/Else" statements to keep the animation looping for the viewer.
+                </p>
+              </div>
+          );
+      }
+  }
+  
+  export default City;
```
Comment (kushagharahi) [inline src/components/p5js-components/city.js:72]: probably missed a lot of formatting that's needed in this

File: src/components/p5js-components/example.js [hunk 40]
```diff
@@ -0,0 +1,31 @@
+export default function sketch(p5) {
```
Comment (kushagharahi) [inline src/components/p5js-components/example.js]: is this example doing anything? looks different than the other p5 react components. this isn't even a react file.

File: src/components/p5js-components/fishtank.js [hunk 41]
```diff
@@ -0,0 +1,112 @@
+import React from 'react';
+import p5 from 'p5';
+import '../projects.css';
+
+
+class Fishtank extends React.Component {
```
Comment (kushagharahi) [inline src/components/p5js-components/fishtank.js:5]: too many carriage returns

**Comment by kushagharahi** [comment]
Formatting love needed, will rereview after you comb through the rest of the files

File: public/index.html (hunk 5 above)
Comment (kushagharahi) [reply public/index.html]: https://developers.google.com/search/docs/beginner/seo-starter-guide

File: .eslintcache (hunk 2 above)
Comment (kushagharahi) [reply .eslintcache]: also delete it from your git cache to git rid of it here (google how to do this)

**Pull request merged**